src/wmtk/operations/tri_mesh/EdgeSplitWithTag.hpp

```diff
@@ -0,0 +1,63 @@
+#pragma once
+#include <wmtk/TriMesh.hpp>
+#include <wmtk/operations/TupleOperation.hpp>
+#include "EdgeSplit.hpp"
+
+namespace wmtk::operations {
+namespace tri_mesh {
+class EdgeSplitWithTag;
+}
+
+enum { TAGS_DIFFERENT, TAGS_SAME };
+
+template <>
+struct OperationSettings<tri_mesh::EdgeSplitWithTag>
+{
+    OperationSettings<tri_mesh::EdgeSplit> split_settings;
+    InvariantCollection invariants;
+    // handle to vertex position
+    MeshAttributeHandle<double> position;
+    MeshAttributeHandle<long> vertex_tag;
+    MeshAttributeHandle<long> edge_tag;
+    MeshAttributeHandle<long> split_todo;
+    long split_vertex_tag_value;
+    long split_edge_tag_value;
+    long embedding_tag_value;
+    bool need_embedding_tag_value;
+    // too short edges get ignored
+    double min_squared_length = -1;
+
+    void initialize_invariants(const TriMesh& m);
+
+    // debug functionality to make sure operations are constructed properly
+    bool are_invariants_initialized() const;
+};
+
+namespace tri_mesh {
+class EdgeSplitWithTag : public TriMeshOperation, private TupleOperation
```

Comment:
Also here, please explain what this operation is doing. 
We have little to no documentation in our code and this is an issue we need to tackle.